iosApp/iosApp/ui/buttons.swift

```diff
@@ -0,0 +1,81 @@
+import SwiftUI
+
+struct TextOnlyButton: View {
+    var text: String
+    var enabled: Bool
+    var textColor: Color
+    var action: () -> Void
+    
+    init(text: String, enabled: Bool = true, textColor: Color = .black, action: @escaping () -> Void) {
+        self.text = text
+        self.enabled = enabled
+        self.textColor = textColor
+        self.action = action
+    }
+    
+    var body: some View {
+        Button(action: action) {
+            Text(text)
+                .foregroundColor(textColor)
+                .padding()
+                .background(Color.clear)
+                .cornerRadius(5)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 5)
+                        .stroke(textColor, lineWidth: 1)
+                )
+        }
+        .disabled(!enabled)
+    }
+}
+
+struct AppButton: View {
+    var text: String
+    var enabled: Bool
+    var textColor: Color
+    var action: () -> Void
+    
+    init(text: String, enabled: Bool = true, textColor: Color = .black, action: @escaping () -> Void) {
+        self.text = text
+        self.enabled = enabled
+        self.textColor = textColor
+        self.action = action
+    }
+    
+    var body: some View {
+        Button(action: action) {
+            Text(text)
+                .foregroundColor(textColor)
+                .padding()
+                .background(enabled ? Color.blue : Color.gray)
+                .cornerRadius(5)
+        }
+        .disabled(!enabled)
+    }
+}
+
+struct IconButton: View {
+    var icon: Image
+    var enabled: Bool
+    var action: () -> Void
+    
+    init(icon: Image, enabled: Bool = true, action: @escaping () -> Void) {
```

Comment:
What does this @escaping mean?

Reply:
From my understanding it allows that "action" parameter to be executed after the init has been called. Since (I think) if we don't have it there, swift will try to execute it immediately. This just ensures that it won't be cleaned up since it might be executed later (on button press for example, where we don't want to call this "action" when we instantiate the button )